### PR TITLE
API for KEB tracing events

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

This is a followup to KEB tracing events proposal in https://github.com/kyma-project/control-plane/pull/2094. Exposing events as a API with some basic filtering options in form of URL param `?instance_id=`, `?runtime_id=`, `?operation_id=`

<details>
<summary>example</summary>

state from the db:
```
broker=> select * from events where instance_id='test-jw231';
                  id                  | level | instance_id |             operation_id             |                          message                          |          created_at
--------------------------------------+-------+-------------+--------------------------------------+-----------------------------------------------------------+-------------------------------
 3b36a6f8-4cf9-44e7-998c-38cc1e60e352 | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Starting]                               | 2022-10-05 07:07:25.313303+00
 23ef8cc2-35b1-4d2e-9f65-27dd561ecaf8 | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Provision_Initialization]               | 2022-10-05 07:07:25.328905+00
 a3aa693d-a354-4a62-9a49-06a4f637d4ca | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Resolve_Target_Secret]                  | 2022-10-05 07:07:25.495446+00
 41fb10d5-cba3-40bf-b2e7-aae2c6b965f1 | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [AVS_Create_Internal_Eval_Step]          | 2022-10-05 07:07:25.611217+00
 d6c2a580-5a9c-4ea8-9b77-f0d8b5ea1abb | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [EDP_Registration]                       | 2022-10-05 07:07:26.206864+00
 77188880-47b9-4c39-a133-c95e5919a92e | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Overrides_From_Secrets_And_Config_Step] | 2022-10-05 07:07:27.195825+00
 c3d2a7ae-b5a3-48f6-80ea-9ddd6c6873be | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [BTPOperatorOverrides]                   | 2022-10-05 07:07:27.230103+00
 dfbd0890-c17f-4544-a212-709cc71e96be | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Create_Runtime_Without_Kyma]            | 2022-10-05 07:07:27.235336+00
 98d6c6cf-37d1-4c42-b6ec-693027107f1a | info  | test-jw231  | 9298b370-af3d-46b6-9bdb-2ae814cdaf21 | processing step: [Check_Runtime]                          | 2022-10-05 07:07:38.589684+00
(9 rows)

```

query on a locally exposed KEB:
```
$ curl -XGET 'localhost:8080/info/events?instance_id=test-jw231' | jq '.'
[
  {
    "ID": "3b36a6f8-4cf9-44e7-998c-38cc1e60e352",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Starting]",
    "CreatedAt": "2022-10-05T07:07:25.313303Z"
  },
  {
    "ID": "23ef8cc2-35b1-4d2e-9f65-27dd561ecaf8",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Provision_Initialization]",
    "CreatedAt": "2022-10-05T07:07:25.328905Z"
  },
  {
    "ID": "a3aa693d-a354-4a62-9a49-06a4f637d4ca",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Resolve_Target_Secret]",
    "CreatedAt": "2022-10-05T07:07:25.495446Z"
  },
  {
    "ID": "41fb10d5-cba3-40bf-b2e7-aae2c6b965f1",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [AVS_Create_Internal_Eval_Step]",
    "CreatedAt": "2022-10-05T07:07:25.611217Z"
  },
  {
    "ID": "d6c2a580-5a9c-4ea8-9b77-f0d8b5ea1abb",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [EDP_Registration]",
    "CreatedAt": "2022-10-05T07:07:26.206864Z"
  },
  {
    "ID": "77188880-47b9-4c39-a133-c95e5919a92e",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Overrides_From_Secrets_And_Config_Step]",
    "CreatedAt": "2022-10-05T07:07:27.195825Z"
  },
  {
    "ID": "c3d2a7ae-b5a3-48f6-80ea-9ddd6c6873be",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [BTPOperatorOverrides]",
    "CreatedAt": "2022-10-05T07:07:27.230103Z"
  },
  {
    "ID": "dfbd0890-c17f-4544-a212-709cc71e96be",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Create_Runtime_Without_Kyma]",
    "CreatedAt": "2022-10-05T07:07:27.235336Z"
  },
  {
    "ID": "98d6c6cf-37d1-4c42-b6ec-693027107f1a",
    "Level": "info",
    "InstanceID": "test-jw231",
    "OperationID": "9298b370-af3d-46b6-9bdb-2ae814cdaf21",
    "Message": "processing step: [Check_Runtime]",
    "CreatedAt": "2022-10-05T07:07:38.589684Z"
  }
]
```
</details>

Depends on:
- [x] https://github.com/kyma-project/control-plane/pull/2094

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/storage/dbmodel/operation.go [hunk 1]
```diff
@@ -49,3 +49,24 @@ type InstanceERSContextStatsEntry struct {
 	LicenseType sql.NullString
 	Total       int
 }
+
+type EventLevel string
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/storage/dbmodel/operation.go:53]: minor: I would move it to separate file, for example event.go

File: components/kyma-environment-broker/internal/storage/postsql/read.go [hunk 2]
```diff
@@ -685,6 +685,19 @@ func (r readSession) ListInstances(filter dbmodel.InstanceFilter) ([]dbmodel.Ins
...
+func (r readSession) ListEvents(filter dbmodel.EventFilter) ([]dbmodel.EventDTO, error) {
+	var events []dbmodel.EventDTO
+	stmt := r.session.Select("*").From("events")
+	if filter.InstanceID != "" {
+		stmt.Where(dbr.Eq("instance_id", filter.InstanceID))
+	}
+	if filter.OperationID != "" {
+		stmt.Where(dbr.Eq("operation_id", filter.OperationID))
+	}
+	_, err := stmt.Load(&events)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/storage/postsql/read.go:697]: did you think about adding `order by` part to the SQL statement? usually we expects ordered events (by created_at)

Comment (wozniakjan) [reply components/kyma-environment-broker/internal/storage/postsql/read.go:697]: good point, I will add that

**Comment by wozniakjan** [comment]
> please update the file: resources/kcp/charts/kyma-environment-broker/templates/authorization-policy.yaml

Added as a standalone `AuthorizationPolicy` in https://github.com/kyma-project/control-plane/pull/2114/commits/3599be76de3f7d650252adbd4d5f21491ce391e1 to follow the convention, but I think in the future, good amount of the file can be consolidated as most of the rules are copy-pasted

File: resources/kcp/charts/kyma-environment-broker/templates/authorization-policy.yaml [hunk 3]
```diff
@@ -143,6 +143,46 @@ spec:
...
+        methods:
+        - GET
+        paths:
+        - /events
+    from:
+    - source:
+        principals:
+{{- with .Values.runtimeAllowedPrincipals }}
+{{ tpl . $ | indent 10 }}
+{{- end }}
```
Comment (wozniakjan) [inline resources/kcp/charts/kyma-environment-broker/templates/authorization-policy.yaml:178]: afaiu this is for metrics, currently not used but maybe desired follow-up to add metrics?

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**